lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py

```diff
@@ -852,7 +852,11 @@ def move_directory_files( self, current_dir, source_dir, destination_dir ):
             os.makedirs(destination_directory)
         for dir_entry in os.listdir(source_directory):
             source_entry = os.path.join(source_directory, dir_entry)
-            if os.path.islink(source_entry):
+            if os.path.isdir(source_entry):
+                destination_subdir = os.path.join(destination_directory, dir_entry)
+                if not os.path.exists(destination_subdir):
+                    os.makedirs(destination_subdir)
+            elif os.path.islink(source_entry):
```

Comment:
So this is my test directory, with one folder with 1 file inside:
```
In [3]: !ls -l -R
.:
total 4
drwxrwxr-x 2 marius marius 4096 juil. 12 17:24 subdir

./subdir:
total 0
-rw-rw-r-- 1 marius marius 0 juil. 12 17:24 test
```

If I call `move_directory_files` like this:
```
move_directory_files(self='', current_dir='.', source_dir='.', destination_dir='/tmp/testdir')
```

I lose the `test` file:
```
In [8]: !ls -l -R /tmp/testdir
/tmp/testdir:
total 4
drwxrwxr-x 2 marius marius 4096 juil. 12 17:29 subdir

/tmp/testdir/subdir:
total 0

```
That's not the desired behaviour, right?
With the previous versrion I do get the test file.